tests/performance/event_generator/src/event_generator/main.py

```diff
@@ -0,0 +1,181 @@
+
+
+def get_logcollector_generator(file_config: dict[str, Any]) -> LogEventGenerator:
+    """Create and return a LogEventGenerator instance configured according to the provided file configuration.
+
+    Args:
+        file_config (dict): Configuration settings specific to the logcollector module. It must include
+                            necessary parameters like module, path, operations, rate, max_file size, and
+                            template path.
+
+    Returns:
+        LogEventGenerator: An instance of LogEventGenerator configured as per the file_config.
+
+    Raises:
+        ValueError: If any required parameters are missing in file_config for logcollector.
+    """
+    required_params = ['module', 'path', 'operations',
+                       'rate', 'max_file_size', 'template_path']
+    if not all(param in file_config for param in required_params):
+        raise ValueError(
+            "Missing required config parameters for logcollector.")
+    return LogEventGenerator(
+        rate=file_config['rate'],
+        path=file_config['path'],
+        operations=file_config['operations'],
+        max_file_size=file_config['max_file_size'],
+        template_path=file_config['template_path']
+    )
+
+
+def get_syscheck_generator(file_config: dict[str, Any]) -> SyscheckEventGenerator:
+    """Create and return a SyscheckEventGenerator instance based on the provided file configuration.
+
+    Args:
+        file_config (dict): Configuration settings specific to the syscheck module. It should include
+                            necessary parameters like module, path, operations, and rate.
+
+    Returns:
+        SyscheckEventGenerator: An instance of SyscheckEventGenerator configured according to the file_config.
+
+    Raises:
+        ValueError: If any required parameters are missing in file_config for syscheck.
+    """
+    required_params = ['module', 'path', 'operations', 'rate']
+    if not all(param in file_config for param in required_params):
+        raise ValueError("Missing required config parameters for syscheck.")
+    return SyscheckEventGenerator(
+        rate=file_config['rate'],
+        path=file_config['path'],
+        operations=file_config['operations']
+    )
+
+
+def main() -> None:
+    """Main function to parse arguments and initiate event generation based on configurations."""
+    args = parse_arguments()
+
+    with open(args.config) as file:
+        config = yaml.safe_load(file)
+
+    threads = []
+    for file_config in config.get('files', []):
+        if file_config['module'] == 'logcollector':
+            generator = get_logcollector_generator(file_config)
+        elif file_config['module'] == 'syscheck':
+            generator = get_syscheck_generator(file_config)
+        else:
+            raise ValueError(
+                "Unsupported module specified in the configuration.")
+
+        thread = threading.Thread(target=generator.start)
+        thread.start()
```

Comment:
Start all the tread once all the processing is over in order to minimize differences in start time between them

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/3cc2e3b11ced84fbedd69cc04f24fe1469b2deb6